pkgs/development/tools/analysis/valgrind/default.nix

```diff
@@ -14,11 +14,13 @@ stdenv.mkDerivation rec {
 
   # Perl is needed for `cg_annotate'.
   # GDB is needed to provide a sane default for `--db-command'.
-  buildInputs = [ perl ] ++ stdenv.lib.optional (!stdenv.isDarwin) gdb;
+  buildInputs = [ perl gdb ]  ++ stdenv.lib.optional (stdenv.isDarwin) [ bootstrap_cmds xnu ];
```

Comment:
Nitpick (also below again): `optional` takes a single element, use `optionals` instead on lists.  Otherwise a nested list is created &ndash; that works but it isn't nice.